overlord/state/state.go

```diff
@@ -165,10 +165,10 @@ func (s *State) writing() {
 
 func (s *State) unlock() {
 	atomic.AddInt32(&s.muC, -1)
-	lockWait, lockStart := s.lockWait, s.lockStart
-	s.lockWait, s.lockStart = 0, 0
+	lockWaitStart, lockHoldStart := s.lockWaitStart, s.lockHoldStart
+	s.lockWaitStart, s.lockHoldStart = 0, 0
 	s.mu.Unlock()
-	osutil.MaybeSaveLockTime(lockWait, lockStart)
+	osutil.MaybeSaveLockTime(lockWaitStart, lockHoldStart, osutil.LockTimestamp())
```

Comment:
shouldn't the LockTimestamp be captured before the Unlock, assuming the Unlock anyway is quick?

Reply:
good point, I made a similar comment on the original code and then forgot to update it.